Makefile

```diff
@@ -52,15 +58,23 @@ func-test:
 .PHONY: cephcsi
 cephcsi:
 	if [ ! -d ./vendor ]; then dep ensure -vendor-only; fi
-	CGO_ENABLED=0 GOOS=linux go build -a -ldflags '$(LDFLAGS) -extldflags "-static"' -o  _output/cephcsi ./cmd/
+	CGO_ENABLED=0 GOOS=linux GOARCH=$(GOARCH) go build -a -ldflags '$(LDFLAGS) -extldflags "-static"' -o  _output/cephcsi ./cmd/
 
 image-cephcsi: cephcsi
 	cp _output/cephcsi deploy/cephcsi/image/cephcsi
-	$(CONTAINER_CMD) build -t $(CSI_IMAGE_NAME):$(CSI_IMAGE_VERSION) deploy/cephcsi/image
+	chmod +x deploy/cephcsi/image/cephcsi
+	$(CONTAINER_CMD) build -t $(CSI_IMAGE) deploy/cephcsi/image
 
 push-image-cephcsi: image-cephcsi
-	$(CONTAINER_CMD) push $(CSI_IMAGE_NAME):$(CSI_IMAGE_VERSION)
-
+	$(CONTAINER_CMD) tag $(CSI_IMAGE) $(CSI_IMAGE)-$(GOARCH)
+	$(CONTAINER_CMD) push $(CSI_IMAGE)-$(GOARCH)
```

Comment:
@cyb70289 want final confirmation before merging this PR, can we push MultiArch images to quay.io? have you tested it

Reply:
@Madhu-1 my free quay.io account does NOT support multiarch image, as we've talked in [this thread](https://github.com/ceph/ceph-csi/issues/671#issuecomment-548271431). I validated multiarch pushing code with my dockerhub account.
Quay.io v3.0 [announces multiarch support](https://access.redhat.com/documentation/en-us/red_hat_quay/3/html/red_hat_quay_release_notes/index#multi_architecture_container_image_support). I don't know if ceph-csi registry is hosted by quay.io v3.0. It's better to try it offline first.

Reply:
Below commands are for a quick test. It pushes multiarch image of busybox to a repo. You may adjust per your requirements.

# set docker registry
export REPO=quay.io/cephcsi

# login registry, input user, pass
docker login quay.io

# push busybox amd64 and arm64 images
docker pull busybox
docker pull arm64v8/busybox

docker tag busybox:latest $REPO/busybox-amd64:latest
docker tag arm64v8/busybox:latest $REPO/busybox-arm64:latest

docker push $REPO/busybox-amd64:latest
docker push $REPO/busybox-arm64:latest

# push multi arch image
export DOCKER_CLI_EXPERIMENTAL=enabled

docker manifest create --amend $REPO/busybox:latest $REPO/busybox-amd64:latest $REPO/busybox-arm64:latest

docker manifest annotate $REPO/busybox:latest $REPO/busybox-amd64:latest --os linux --arch amd64
docker manifest annotate $REPO/busybox:latest $REPO/busybox-arm64:latest --os linux --arch arm64

docker manifest push $REPO/busybox:latest

Reply:
@humblec can you please test above one?